pennylane/devices/null_qubit.py

```diff
@@ -277,10 +278,28 @@ def _max_workers(self):
         """No-op property to allow for borrowing DefaultQubit.preprocess without AttributeErrors"""
         return None
 
+    # pylint: disable=cell-var-from-loop
     def preprocess(
         self, execution_config=DefaultExecutionConfig
     ) -> Tuple[TransformProgram, ExecutionConfig]:
         program, _ = DefaultQubit.preprocess(self, execution_config)
+        for t in program:
+            if t.transform == decompose.transform:
+                original_stopping_condition = t.kwargs["stopping_condition"]
+
+                def new_stopping_condition(op):
+                    return (not op.has_decomposition) or original_stopping_condition(op)
```

Comment:
Is there any case where the original stopping condition yields False for some reason not related to the decomposition that we then override for ops with decompositions? Not sure how common that edge case is?

Reply:
For example, null.qubit will now support things like:
```
class MyRandomOp(qml.operation.Operator):
    pass
```

that doesn't provide any additional definition.  `default.qubit` wouldn't let such an operation through preprocessing because it would cause the execution to fall over in strange and unexpected ways.

Reply:
No I mean, something that would have not decomposed on default qubit for some other reason might now decompose on null.qubit if it has a decomposition. 

I guess it's a bit of an edge case we don't care about. Im fine with keeping it as is 👍🏼

Reply:
if `original_stopping_condition(op)` is `True`, the `new_stopping_condition(op)` will also be `True`